test/integration/crud/insert.test.js

```diff
@@ -428,25 +428,21 @@ describe('crud - insert', function () {
         requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
       },
 
-      test: function (done) {
-        var configuration = this.configuration;
-        var client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
-        client.connect(function (err, client) {
-          var db = client.db(configuration.db);
-          var collection = db.collection('test_to_json_for_long');
-
-          collection.insert(
-            [{ value: Long.fromNumber(32222432) }],
-            configuration.writeConcernMax(),
-            function (err, ids) {
-              test.ok(ids);
-              collection.findOne({}, function (err, item) {
-                test.equal(32222432, item.value);
-                client.close(done);
-              });
-            }
-          );
-        });
+      test: async function () {
+        const configuration = this.configuration;
+        const client = configuration.newClient(configuration.writeConcernMax(), { maxPoolSize: 1 });
+        await client.connect();
+        this.defer(async () => {
```

Comment:
All the cases in these tests that defer a `client.close()` are not necessary - we can just `await client.close()` at the end of the tests.